design/cli-install-changes.md

```diff
@@ -0,0 +1,286 @@
+```
+      --backup-location-config mapStringString     configuration to use for the backup storage location. Format is key1=value1,key2=value2
+      --bucket string                              name of the object storage bucket where backups should be stored
+      --prefix string                              prefix under which all Velero data should be stored within the bucket. Optional.
+      --dry-run                                    generate resources, but don't send them to the cluster. Use with -o. Optional.
+```
+
+...`snapshot-location create`
+```
+      --snapshot-location-config mapStringString   configuration to use for the volume snapshot location. Format is key1=value1,key2=value2
+```
+
+...both `backup-location create` and `snapshot-location create`
+```
+      --provider string                            provider name for backup and volume storage 
+      --no-secret                                  flag indicating if a secret should be created. Must be used as confirmation if --secret-file is not provided. Optional.
+      --secret-file string                         file containing credentials for backup and volume provider. If not specified, --no-secret must be used for confirmation. Optional.
+      --dry-run                                    generate resources, but don't send them to the cluster. Use with -o. Optional.
+```
+
+...`restic`
+```
+      --default-restic-prune-frequency duration    how often 'restic prune' is run for restic repositories by default. Optional.
+      --restic-pod-cpu-limit string                CPU limit for restic pod. A value of "0" is treated as unbounded. Optional. (default "0")
+      --restic-pod-cpu-request string              CPU request for restic pod. A value of "0" is treated as unbounded. Optional. (default "0")
+      --restic-pod-mem-limit string                memory limit for restic pod. A value of "0" is treated as unbounded. Optional. (default "0")
+      --restic-pod-mem-request string              memory request for restic pod. A value of "0" is treated as unbounded. Optional. (default "0")
+      --use-restic                                 create restic deployment. Optional.
+```
+
+...`plugin`
+ ```
+      --plugins stringArray                        Plugin container images to install into the Velero Deployment
+      --sa-annotations mapStringString             annotations to add to the Velero ServiceAccount. Add iam.gke.io/gcp-service-account=[GSA_NAME]@[PROJECT_NAME].iam.gserviceaccount.com for workload identity. Optional. Format is key1=value1,key2=value2
+```
+
+##### Velero Server
+`velero server (DEPRECATED)` 
+ 
+`velero server --default-backup-storage-location (DEPRECATED)` moved to `velero backup-location set --default` 
+
+`velero server --default-volume-snapshot-locations (DEPRECATED)` moved to `velero snapshot-location set --default`
+
+`velero server --default-restic-prune-frequency (DEPRECATED)` moved to `velero restic set --default-prune-frequency` 
```

Comment:
Where would this be stored if moved? As a command line flag on `velero server`, this is stored within the command line section of the Deployment YAML right now.

Reply:
It would all be stored in the same section of the same resources. If a user decides to configure Velero to use restic, it would have to be after `velero init` and the restic bits would be a patch to the Velero deployment. 

If there's a risk I'm missing something please let me know Nolan!

Reply:
I think the main takeaway is that this flag needs to exist both on the user command (e.g. `velero init`) *and* on the server command.  When the user specifies it via `velero init --default-restic-prune-frequency <value>`, the result of that is that Deployment YAML gets created whose command is `velero server --default-restic-prune-frequency <value>`.

Reply:
Got it. Thanks for the additional clarification, this helps.